src/exporter/DaemonMetricCollector.cc

```diff
@@ -98,91 +98,104 @@ void DaemonMetricCollector::dump_asok_metrics() {
   }
   auto prio_limit = g_conf().get_val<int64_t>("exporter_prio_limit");
   for (auto &[daemon_name, sock_client] : clients) {
-    bool ok;
-    sock_client.ping(&ok);
-    if (!ok) {
-      failures++;
-      continue;
-    }
-    std::string counter_dump_response =
-      asok_request(sock_client, "counter dump", daemon_name);
-    if (counter_dump_response.size() == 0) {
+    try {
```

Comment:
how about extract the body of the loop into another function, probably with the same level of indentation, to avoid the large code churn? so that we can catch the exception on the caller side of this new function. also, i agree that the `exception` type is way too general. better off only catching the exception(s) we are interested in.

Reply:
i've added runtime_error and invalid_argument, that we might expect. i've not refactored the function into a separate function, might be out of scope of this pr. 
but i think this PR might be working on refactoring some parts of this code, https://github.com/ceph/ceph/pull/54768

Reply:
well, i don't feel strong. but sometime, i think we need to do refactor to prepare for a simpler change. and now the code suffers from two things

- large code churn
- wrong indent